src/Maker/MakeDto.php

```diff
@@ -0,0 +1,250 @@
+            ->addArgument('bound-class', InputArgument::REQUIRED, 'The name of Entity that the DTO will be bound to')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeDto.txt'))
+        ;
+
+        $inputConf->setArgumentAsNonInteractive('bound-class');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        if (null === $input->getArgument('bound-class')) {
+            $argument = $command->getDefinition()->getArgument('bound-class');
+
+            $entities = $this->entityHelper->getEntitiesForAutocomplete();
+
+            $question = new Question($argument->getDescription());
+            $question->setValidator(function ($answer) use ($entities) {return Validator::existsOrNull($answer, $entities); });
+            $question->setAutocompleterValues($entities);
+            $question->setMaxAttempts(3);
+
+            $input->setArgument('bound-class', $io->askQuestion($question));
+        }
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $dataClassNameDetails = $generator->createClassNameDetails(
+            $input->getArgument('name'),
+            'Form\\Data\\',
+            'Data'
+        );
+
+        $boundClass = $input->getArgument('bound-class');
+
+        $boundClassDetails = $generator->createClassNameDetails(
+            $boundClass,
+            'Entity\\'
+        );
+
+        // get some doctrine details
+        $doctrineEntityDetails = $this->entityHelper->createDoctrineDetails($boundClassDetails->getFullName());
+
+        if (null === $doctrineEntityDetails) {
+            $io->error([
+                'The bound class is not a valid doctrine entity.',
+            ]);
+
+            return;
+        }
+
+        // get class metadata (used by regenerate)
+        $metaData = $this->entityHelper->getMetadata($boundClassDetails->getFullName());
+
+        // list of fields
+        $fields = $metaData->fieldMappings;
+
+        $boundClassVars = [
+            'bounded_full_class_name' => $boundClassDetails->getFullName(),
+            'bounded_class_name' => $boundClassDetails->getShortName(),
+        ];
+
+        // the result is passed to the template
+        $addHelpers = $io->confirm('Add helper extract/fill methods?');
+
+        // Generate getters/setters
+        $omitGettersSetters = $io->confirm('Omit generation of getters/setters?');
+
+        // filter id from fields
+        $fields = array_filter($fields, function ($field) {
+            // mapping includes id field when property is an id
+            if (!empty($field['id'])) {
+                return false;
+            }
+
+            return true;
+        });
+
+        $DTOClassPath = $generator->generateClass(
+            $dataClassNameDetails->getFullName(),
+            __DIR__.'/../Resources/skeleton/dto/Data.tpl.php',
```

Comment:
I'd call the template `DTO.tpl.php` - just so it's easier to recognize in the library.